task3-4/task3-4/MainScreen/MainScreenViewController.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  ViewController.swift
+//  task3-4
+//
+//  Created by Григоренко Александр Игоревич on 23.09.2022.
+//
+
+import UIKit
+
+/// VC экрана с заказом
+class MainScreenViewController: UIViewController {
+    private var goToCheckScreen: UIAlertController = UIAlertController()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    @IBAction func goToCheckScreen(_ sender: Any) {
+        goToCheckScreen = UIAlertController(title: " Выставить чек? ", message: nil, preferredStyle: .actionSheet)
+        goToCheckScreen.addAction(UIAlertAction(title: "Cancel", style: .default))
+        goToCheckScreen.addAction(UIAlertAction(title: "Чек", style: .default, handler: { _ in
```

Comment:
```suggestion
        goToCheckScreen.addAction(UIAlertAction(title: "Чек", style: .default, handler: { [weak self] in
```
 похоже на твой случай

https://stackoverflow.com/questions/34931172/should-self-be-captured-as-strong-in-a-uialertactions-handler

The key question to ask yourself is if your alert object is "owned" by self. In this case, it is not (because you declared let alert = ... in the function body). So you do not need to create this as a weak or unowned reference.

If alert was a property of self, then it would be "owned" by self and that is when you would want to create a weak reference to self in the closure "owned" by the alert.